api/authCustom/views.py

```diff
@@ -0,0 +1,30 @@
+from rest_framework.decorators import api_view, parser_classes
+from rest_framework.parsers import JSONParser
+from rest_framework.response import Response
+from django.utils import timezone
+
+from .models import Token_Custom
+from .utils import env
+
+
+
+@api_view(["POST"])
+@parser_classes([JSONParser])
+def goal_auth_token(request, userId):
+    try:
+        rest_key = request.data.get('rest_key')
+        if rest_key != env('REST_KEY'):
+            raise Exception("Wrong REST_KEY")
+
+        try:
+            token = Token_Custom.objects.get(userId=userId)
+            if token.is_invalid(createdTime=token.created.timestamp()):
+                Token_Custom.objects.filter(userId=userId).update(key=Token_Custom.generate_key(), created=timezone.now())
+                token = Token_Custom.objects.get(userId=userId)
+        except Token_Custom.DoesNotExist:
+            token = Token_Custom(userId=userId)
+            token.save()
```

Comment:
Ideally, the view layer shouldn't concern itself with if the token, (or any data) is present or not. It's the responsibility of model layer.

Reply:
explain.